web/static/models/transfer-model.js

```diff
@@ -0,0 +1,89 @@
+import * as util from "../util.js";
+
+/**
+ * Model class for transfer dialog data and state
+ */
+export class TransferModel {
+    /**
+     * @param {number} mode - Transfer mode (GET/PUT)
+     * @param {Array<Object>} records - Data records
+     */
+    constructor(mode, records) {
+        this.mode = mode;
+        this.records = records || [];
+        this.selectedIds = new Set();
```

Comment:
When you pick a particular datastructure like this it can be useful to give an explanation of why in the comments. In this case I think it is pretty obvious, you want to keep the selectedIds Unique.